### DISCUSSION TITLE
8bitdo Ultimate BT compatibility

### DISCUSSION THREAD

**Discussion by HENRCAST** [darthcloud/BlueRetro#539] [Q&A] [answered]
Hello! I have a question regarding compatibility with 8Bitdo's Ultimate BT controller (https://www.8bitdo.com/ultimate-bluetooth-controller/). 

Unlike the PRO2 controller, the Ultimate BT does not support X-Input through BT, but only through the 2.4g dongle. When you connect it to a PC through BT, it is recognized as a Switch Pro Controller. However, Blueretro apparently only works with 8Bitdo controllers via X-input.

So, two questions:
- Can the Ultimate BT connect to Blueretro at all?
- If yes, does it retain analogue triggers, or are the triggers only digital (like Switch controllers)?

Thank you!

**Comment by darthcloud** [comment, marked as answer]
I don't have one, so I can't tell you.

If connected as a switch controller it will only have digital trigger.

Regardless of their documentation, in Bluetooth mode if you power on the controller with X + Start (or whatever is the power on button) and then you press Sync does the controller pair to windows as a xbox controller (or as 8bitdo one) instead of switch?

If they removed this it's the ultimate fail.

**Reply by HENRCAST** [reply]
I don't own one either, but I read from others users that the button holding trick for X-input no longer works* - Bluetooth connection really only supports Switch mode. I guess then that the "Ultimate" controller would connect to Blueretro, but lose analog trigger input.

I was planning one getting one to use with my Switch and Gamecube, but I guess I will have to look for an alternative.

*see kavika's comment in this discussion: https://steamcommunity.com/app/1675200/discussions/2/3472865401883691481/

**Reply by thiago23** [reply]
8bitdo ultimate control does not work on the esp32, it connects but does not receive input, the xbox one control works normally, the console is the ps2

**Comment by darthcloud** [comment]
They got like 3 different Ultimate controller now, They should all work in latest beta:
https://github.com/darthcloud/BlueRetro/actions/runs/13083210355

**Reply by chopsticksamurai** [reply]
I wanted to mention I just picked up one of these controllers today, found this thread, and the beta firmware works great.  Thanks so much for adding support!

**Reply by gromber** [reply]
yeah tested with 8Bitdo Ultimate 2.4G Wireless Controller in bluetooth dinput mode with that beta without problem, maybe thumbstick settings need to be improved.

But now, rumble is disabled in xbox one bt gamepad that worked in lastest HW1 GC stable installed on a Wii Family.

I must activate native controls in nintendont to work.

**Reply by lrdeavila** [reply]
> They got like 3 different Ultimate controller now, They should all work in latest beta: https://github.com/darthcloud/BlueRetro/actions/runs/13083210355

Waiting for firmware update here https://github.com/RetroScaler/NGC-BlueRetro/releases

PLZPLZPLZ!!! THX

**Reply by darthcloud** [reply]
Ask on the RetroScaler repo not here

**Reply by lrdeavila** [reply]
> Ask on the RetroScaler repo not here

Sorry, it mentions git there, I thought it had some connection.
